src/components/Heading/Heading.tsx

```diff
@@ -25,52 +28,69 @@ export type HeadingTypes =
 
 export type HeadingTagTypes = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span' | 'legend'
 
-type ElementProps = Omit<HTMLAttributes<HTMLElement>, keyof Props | keyof TextProps>
+type ElementProps = Omit<
+  HTMLAttributes<HTMLElement>,
+  keyof Props | keyof TextProps | 'role' | 'aria-level'
+>
+
+const generateTagProps = (tag: HeadingTagTypes | undefined, level: number) => {
+  const forwardedAs = tag || ((level <= 6 ? `h${level}` : 'span') as HeadingTagTypes)
+  let role = undefined
+  let ariaLevel = undefined
```

Comment:
NITS: 意図的でなければ明示的な初期化不要そう（なぜ明示的に undefined を入れてるのか気になってしまう

```suggestion
  let role
  let ariaLevel
```

Reply:
この関数はコンポーネント内部からしか呼ばれないため、引数が存在しないわけではなく、明確にundefinedが渡されるので意図的にundefinedを型指定しています〜

Reply:
`let role` でも role = undefined になりませんか?

Reply:
`let role` でも undefinedが入りますがrole属性は level > 6 の場合にだけ、表示し、それ以外は非表示にしたい、という意図があるため、最終的にhtmlに出力されないようundefinedを指定している、ということを明示するためにも`= undefined` は宣言したいかなぁ....
そもそもコンパイルしたりする際にuknmrさんが書いたようなコードに圧縮される気がするため、 `let role = undefined` と書いてもデメリットは無いはず。